src/client/api/pythonApi.ts

```diff
@@ -107,7 +115,7 @@ export class PythonExtensionChecker implements IPythonExtensionChecker {
         return this.extensions.getExtension(this.pythonExtensionId) !== undefined;
     }
 
-    public async showPythonExtensionInstallRequiredPrompt(): Promise<void> {
+    public async showPythonExtensionInstallRequiredPrompt(): Promise<boolean | void> {
         if (this.waitingOnInstallPrompt) {
             return this.waitingOnInstallPrompt;
```

Comment:
I'd try to go with two states instead of an enum. And ensure we return the right value here as well.